targets/simple_switch/main.cpp

```diff
@@ -20,15 +20,59 @@
 
 /* Switch instance */
 
+#include <dlfcn.h>
 #include <bm/SimpleSwitch.h>
 #include <bm/bm_runtime/bm_runtime.h>
 #include <bm/bm_sim/target_parser.h>
 
+#include <sstream>
+#include <string>
+#include <vector>
+
 #include "simple_switch.h"
 
 namespace {
 SimpleSwitch *simple_switch;
-bm::TargetParserBasic *simple_switch_parser;
+
+std::string load_modules_option = "load-modules";
+
+class TargetParserModules : public bm::TargetParserBasic {
```

Comment:
I'm not sure why inheritance is needed here. It seems you can just use `TargetParserBasic` as it was intended:
```
simple_switch_parser->add_string_option(load_modules_option, "load the given .so files as modules");
std::string modules;
auto retval = get_string_option(load_modules_option, &modules);
...
load_modules(modules);
```
And if you do think it is better to create a new object that inherits from `TargetParserBasic`, I think it would make sense to take care of ALL target-specific options there.

Reply:
> I'm not sure why inheritance is needed here. It seems you can just use `TargetParserBasic` as it was intended:

The modules have to be loaded before the switch is initialized. simple_switch verifies the loaded json file. If it has external calls that are provided by the module, sinple_switch will exit unless the module is loaded first.

By overriding `TargetParserBasic`, I am able to inject code after the command line arguments are parsed, but before the switch is initialized.

> And if you do think it is better to create a new object that inherits from TargetParserBasic, I think it would make sense to take care of ALL target-specific options there.

I'll do that.